test_build_tag_push.py

```diff
@@ -0,0 +1,99 @@
+                         'app.yml')
+
+
+class TestBuildTagPush(unittest.TestCase):
+
+    def test_parse_app_file(self):
+
+        data_from_app_yml = build_tag_push.parse_app_file(path_to_test_yml)
+        self.assertEqual(data_from_app_yml['name'], 'build-setup-test')
+        self.assertEqual(data_from_app_yml['build'], 'build.sh')
+        self.assertEqual(data_from_app_yml['test'], 'test.sh')
+        self.assertEqual(data_from_app_yml['docker_context'],
+                         ['build', 'static', 'Dockerfile'])
+
+    def test_build_check_app_throws_on_build(self):
+        data_from_app_yml = build_tag_push.parse_app_file(path_to_test_yml)
+
+        with self.assertRaises(
+             build_tag_push.BuildFailedException) as shouldThrow:
+
+            build_tag_push.build_check_app(data_from_app_yml,
+                                           path_to_fail_test_yml, True)
+
+        self.assertEqual("The build failed.", shouldThrow.exception.message)
+
+    def test_build_check_app_runs_build(self):
+        data_from_app_yml = build_tag_push.parse_app_file(path_to_test_yml)
+        build_tag_push.build_check_app(data_from_app_yml, path_to_test_yml,
+                                       True)
+        self.assertTrue(os.path.exists('/Users/alyssakelley/Desktop/MVPStudio/'
+                        'build_tag_push/test_data/builds_succeeds_example/tmp/'
+                                       'build_done.sh'))
+
+    def test_create_context_directory(self):
+
+        context_array = ['build', 'static', 'Dockerfile']
+        test_path = build_tag_push.create_context_directory(context_array,
+                                                            path_to_test_yml)
+        output = os.listdir(test_path)
+        self.assertEqual(output, ['Dockerfile', 'static', 'build'])
+
+    def test_build_docker_image(self):
+        data_from_app_yml = build_tag_push.parse_app_file(path_to_test_yml)
+
+        capturedOutput = io.StringIO()
+        sys.stdout = capturedOutput
+        build_tag_push.build_docker_image(data_from_app_yml, path_to_test_yml)
+        sys.stdout = sys.__stdout__
+
+        self.assertEqual(capturedOutput.getvalue(), 'Building the docker '
+                         'image for build-setup-test\nCreating a '
+                         'temporary directory containting [\'build\', '
+                         '\'static\', \'Dockerfile\']\n')
+
+    def test_tag_docker_image(self):
+        name_from_app_yml = 'build-setup-test'
+        docker_hub_name = 'alyssakelley'
+
+        test_current_hash = subprocess.check_output(
+                            ["git", "log", "-1", "--pretty=%h"]).strip(
+                                ).decode('ascii')
+
+        capturedOutput = io.StringIO()
+        sys.stdout = capturedOutput
+        build_tag_push.tag_docker_image(name_from_app_yml, docker_hub_name)
+        sys.stdout = sys.__stdout__
+
+        self.assertEqual(capturedOutput.getvalue(), "Tagging the docker "
```

Comment:
as above, instead of doing the stdout thing you can check that the image now has the desired tag. And as above be sure to delete the image first. Also: this assumes the image already exists and it usually does because the test before this created it, but tests should be independent: you can, and often want to, run just a single test and some test frameworks randomize test order to ensure you're not accidentally relying on some state from a previous test. Here you can call the `build_docker_image` function again - if the image already exists that won't do anything but if it doesn't it'll get created ensuring your test is run with the right preconditions.